### PR TITLE
repo-updater: Use GitHub REST API to list viewer repositories

### PR DESCRIPTION
**Pull request by keegancsmith**
The GraphQL API currently has a regression where it doesn't respect the
affiliated parameter.

Fixes https://github.com/sourcegraph/sourcegraph/issues/440

### CODE REVIEW

**Comment by keegancsmith** [comment]
@sqs why are we using the graphql API for github?

Can both of you test this change locally. I've done some testing but want this is critical logic.

File: cmd/repo-updater/internal/externalservice/github/repos.go [hunk 1]
```diff
@@ -301,53 +301,16 @@ func (c *Client) ListPublicRepositories(ctx context.Context, sinceRepoID int64)
 // ListViewerRepositories lists GitHub repositories affiliated with the viewer (the currently authenticated user).
 // The nextPageCursor is the ID value to pass back to this method (in the "after" parameter) to retrieve the next
```
Comment (nicksnyder) [inline cmd/repo-updater/internal/externalservice/github/repos.go]: this doc needs to be updated

**Comment by ggilmore** [comment]
> @sqs why are we using the graphql API for GitHub?

Is there a downside w/ preferring the newer API? Obviously, with this issue there was a regression, but I don't see why we wouldn't prefer the latest version (all things being equal).

**Comment by sqs** [comment]
@keegancsmith @ggilmore 

> @sqs why are we using the graphql API for github?

IIRC it's a historical reason. We had some code (eg for fetching all visible repos to a user) from when we offered private code support on Sourcegraph.com that needed to use the GraphQL API because only that was supported for GitHub Apps. It also seemed like GitHub was not adding new features to the REST API and all new stuff would be offered in GraphQL. (That turned out to be the opposite here!)

There are no ongoing reasons to prefer one or the other AFAIK, beyond what @ggilmore mentioned.

**Comment by nicksnyder** [comment]
GitHub confirmed that there was a change in behavior in the API and we would need to update our GraphQL query to make it continue working: https://platform.github.community/t/unable-to-fetch-users-repositories-by-organization-membership/7557/4

Since I assume that we need this code path to work simultaneously on github.com and GitHub Enterprise, it is probably best to merge this and do a patch release tomorrow.

**Pull request merged**